### PR TITLE
Shared daemon wire client (improvement plan 3a)

### PR DESCRIPTION
**Pull request by lucascaro**
Phase 3a of `docs/analysis/2026-07-19-improvement-plan` — the plan's top-ranked remaining item.

## Problem
The daemon wire-client logic (dial+HELLO/WELCOME handshake, write-mutex conn wrapper, control/attach read-loop dispatch) existed **three times**: `cmd/hivegui/app.go`, `cmd/hived-ws-bridge/main.go`, and `internal/wire/testclient`. The two production copies had zero tests, and protocol changes needed three synchronized edits.

## Change
- **`internal/wire.Client`** — the one shared client: handshake (closes conn on failure, fills protocol version, preserves non-zero versions for rejection tests), mutex-serialized `WriteJSON`/`WriteFrame`, single-reader `ReadFrame`, mutex-free `Close`.
- **`wire.ControlEventName` / `wire.AttachEventName`** — the frame→UI-event dispatch tables. GUI (Wails events) and bridge (WS notifications) emit identical event names, so a new fanout frame is added once and both pick it up.
- GUI/bridge keep only app-specific glue: spawn-on-miss dialing, emit targets, the attach burst probe. `testclient` keeps its public API and channel demux but delegates the protocol layer.

## Test-first, per the plan
- New `internal/wire/client_test.go`: handshake happy/refusal/unexpected-frame/deadline paths + a concurrency test proving frames can't interleave mid-write.
- `testclient`'s existing tests and the `cmd/hived` Go e2e suite ran **unchanged** as the characterization net.

## Validation
- `go build/vet/test ./...` green
- `go test -tags=e2e ./cmd/hived/...` against a real spawned hived: green
- Playwright **real** suite (real hived through the migrated bridge): green (2 passed, 10 skipped — matches main's phase-1 quarantine state)

Net: production code −12 lines with the three copies collapsed into one 130-line implementation; +190 lines of new unit tests for the previously untested client logic.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Improvements**
  * Unified transport handling across the web bridge, GUI, and client tools with a shared wire-mode protocol client for reliable JSON/frame messaging.
  * Standardized the handshake flow (including timeout behavior and daemon refusal messaging) and improved control/session event dispatch using consistent event-name mapping.
  * Updated session resizing and lifecycle handling to use the negotiated connection context.
* **Bug Fixes**
  * Improved connection cleanup and shutdown behavior on setup failures.
  * Ensured concurrent outgoing frames do not interleave.
* **Tests**
  * Added coverage for handshake edge cases, event-name mapping, and concurrent write integrity.

### CODE REVIEW

File: cmd/hived-ws-bridge/main.go [hunk 1]
```diff
@@ -456,38 +420,16 @@ func (s *session) attachWriteJSON(id string, t wire.FrameType, v any) error {
...
-	if ft != wire.FrameWelcome {
-		_ = conn.Close()
-		return nil, wire.Welcome{}, fmt.Errorf("unexpected frame %s", ft)
+		return nil, fmt.Errorf("dial %s: %w", sockPath, err)
 	}
-	var w wire.Welcome
-	_ = json.Unmarshal(payload, &w)
-	return conn, w, nil
+	return wire.Handshake(conn, hello)
 }
```
[the remark this answers is not in the document]
Comment (lucascaro) [reply cmd/hived-ws-bridge/main.go:435]: Fixed in 511296aed6a039e0466bd305e6a17bad5acfb8d0 — bounded once inside wire.Handshake (5s read deadline, cleared on success) so all three callers get it; testclient's own deadline handling became redundant and was removed.
